bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/ZkLedgerUnderreplicationManager.java

```diff
@@ -854,6 +855,28 @@ public int getLostBookieRecoveryDelay() throws UnavailableException {
         }
     }
 
+    @Override
+    public void notifyUnderReplicationLedgerChanged(GenericCallback<Void> cb) throws UnavailableException {
+        LOG.debug("notifyUnderReplicationLedgerChanged()");
+        Watcher w = new Watcher() {
+            @Override
+            public void process(WatchedEvent e) {
+                if (e.getPath().contains("urL") && e.getType() == Event.EventType.NodeDeleted) {
```

Comment:
checking the event type is cheaper than the contains, I suggest to invert the conditions order

I think it is better to use `getLedgerId(String path)` method instead of the String contains